code/go/0chain.net/chaincore/chain/config.go

```diff
@@ -95,6 +110,93 @@ func NewConfigImpl(conf *ConfigData) *ConfigImpl {
 	return &ConfigImpl{conf: conf}
 }
 
+// This is to provide access to chain configuration from packages which don't have
+// direct access to 'chain' package. Instead they can read the configuration using
+// 'conf' package. example: conf.Configuration.ReadValue("BlockSize")
+func (c *ConfigImpl) ReadValue(name string) (interface{}, error) {
```

Comment:
The value of Miner had been checked to examine if fee is enabled before I touch the code. It was strange to me too but this was there before I change it.

About the ConfigReader interface:
The problem is we need to access the chain configuration from many packages. To read this config we need to add a reference to 'chain' package. But we cannot do that. Since we are reading the chain config from packages which already added as a reference to chain package (either directly or indirectly), so adding chain as a reference to these packages creates a `circular dependency issue`. 

There was different workaround to solve this problem. Best one was to move the chain configuration to a package accessible from all (or most) of the packages. However this demands a lot of changes and I guess we don't want to do this at the time (We are close to Mainnet).

So I solved the problem by adding a ConfigReader Interface in `chaincore/config` package. 

`type ConfigReader interface {
	ReadValue(name string) (interface{}, error)
}`

then I changed the ConfigImpl to implement this interface. And I defined a ChainConfig field to be of type ConfigReader interface in Config struct in `chaincore/config` package to hold the ConfImpl instance. And I assign it at the time of creating a chain instance. 

Using this approach we can access Chain Configuration by accessing to  `chaincore/config` package. 
`conf.Configuration().ReadValue("BlockSize")`